### ISSUE TITLE
GetNumItems

### ISSUE THREAD

**Issue by Pekkape** [llde/xOBSE#84]
I use v21.8

Very odd. When I use GetNumItems it return 1 if the container is empty so I did a check:

`Let NumberOfItemsKeep := PekKeepOneItemRef.GetNumItems

Let TempObject := PekKeepOneItemRef.GetInventoryObject NumberOfItemsKeep

Printc "NumberOfItems = %.0f - NotIncludeObject = %i %n" NumberOfItemsKeep TempObject TempObject
`
Here is the result:
NumberOfItems = 1 - NotIncludeObject = 00000000 <no name>

-------------------------------
Number of Items should be 0. Iy might counted the chest, as I guess getinventoryobject seems do that time to time...  ;)

**Comment by katawful** [comment]
Possibly related to #70?

**Comment by Pekkape** [comment]
Maybe. I am making a dynamic sorter right now, that can mover anything, anywhere and I do want it to work perfect before releaing it for sure...  ;)   I will add more checks in my code.  Thanks a lot katawful to point that one out.

**Comment by katawful** [comment]
The two options are to add a check to make sure this "ghost" item doesn't affect your scripts *or* wait for the next release which fixes it and just for xOBSE usage

**Comment by Pekkape** [comment]
Of course I add a check and correct it. Thanks Kat. So it is an OBSE bug then? I did notice odd behavior long before I used OBSE but did find it first now. Why?  Well the Offset of course. Earlier I did set that variable with set in older scripts but as OblivionAddicted told me to use Let as soon as I use OBSE functions, I suddenly saw it in console while debugging. In my old scripts, it was not a problem, well except when the check failed, the script blamed the player for not follow the instructions, even if the player did follow the instructions. Drake taught me about offset and how to use it and that is so damn brilliant for debugging.

**Comment by katawful** [comment]
The original implementation of inventory references (which is how you're able to work with items in a container) was simply broken and wasn't caught while OBSE was in development. llde rewrote it for the next release and it works without leaving any ghosted items now. Your scripts might mess up if they factor in this ghosted item, so be aware

**Comment by llde** [comment]
@Pekkape upload here the esp so I can test

**Comment by Pekkape** [comment]
That mod is 250 Mb in size....   ;)   Are you sure you want to try it??  ;)  But I can ignore all objects. The spells is for sale at my own merchants, so it might be best to upload it complete?  It is not a stand alone.

**Comment by Pekkape** [comment]
I have not added the workaround yet, which I did plan to do this evening, but if I do, it gets worthless for debugging, so I keep it as it is so you can use the offset

**Comment by llde** [comment]
I'm interested in the ESP only.

**Comment by Pekkape** [comment]
Oki. Well I did packed it with 7z right now. _(failed)_ I upload both so you can choose. :) _(failed)_   The house is located just North of Anvil and a quest will help you get there as well. The script name where the let now reports an offset in is: PekSortingStuffFUN as I did make workarounds in the other scripts connected. It is 4 scripts, working and pointing at each others, so getting them inside ECS wasn't easy in the first place, without remarking them   :D

The function is called from scn PekChestSortingQUSTSCR that is inside the quest scn PekChestSortingQUST

[hobbitHome - esp only.zip](https://github.com/llde/xOBSE/files/5976201/hobbitHome.-.esp.only.zip)

This site only accept 10Mb max size anyway...  It is a zip as this site do not accept 7z

//Pekka

**Comment by Pekkape** [comment]
By the way Idle and Kat. I must add that is is so damn fun that you made xOBSE and continue to develop OBSE. We, well I could most likely speak for the entire Oblivion mod community I guess as it is MUCH appreciated. You rock!!!  :D

//Pekka

**Comment by katawful** [comment]
It's all llde, I'm just the occasional bothersome bug tester

**Comment by Pekkape** [comment]
That's so cool anyway Kat.  :)

Idle: we are making wild guesses and I do not know if they are helpful at all. When we think aboput what is actually happening, we do suspect it is an engine bug and that OBSE finds it and reports it as a bug. As this is not possible to do without OBSE in the first place, so therefor we do find a bug that Bethesda would never be able to find.  This is pure speculation of course and a wild guess, nothing else.

**Comment by Pekkape** [comment]
Even if we solve this, the problem will remain in OBSE and I must keep my fixes in mu code I guess, making it less optimal but I could also make xOBSE an requirement, which will also spread the knowledge about its existence. It must be spread.  ;)  I do mention it everywhere I can.  :)  I am so impressed.

**Comment by llde** [comment]
You may check for specific version of xOBSE or above and complain with a message if the version is lower.
Considering also the amount of stability improval of the first two xOBSe versions (by fixing events mainly)

**Comment by Pekkape** [comment]
Yes, I saw that.  :)   It is odd though that some of the more brilliant ones in our community have not switched to xOBSE yet.

**Comment by llde** [comment]
Ok I checked a bit:
I can't reproduce the issue reported with GetNumItems, It count hidden quest item too, as it should, but I don't have other anomalous beheviours.
The GetInventoryObject instead is exaclty how it should. To acces the first element of the inventory you should do GetInventoryObject 0 and not GetInventoryObject 1. I can't exclude a bug in the function itself, but this isn't one.
